akaze/src/image.rs

```diff
@@ -61,7 +61,21 @@ impl GrayFloatImage {
                     Luma([f32::from(gray_image[(x, y)][0]) / 65535f32])
                 })
             }
-            _ => unreachable!(),
+            other_image => {
+                let color = other_image.color();
+                info!(
+                    "Loaded and converted a {} x {} {:?} image",
+                    input_image.width(),
+                    input_image.height(),
+                    color,
+                );
+                let bits_per_component =
+                    color.bits_per_pixel() as u32 / color.channel_count() as u32;
+                let max_value = (1 << bits_per_component) - 1;
```

Comment:
correct me if I'm wrong but this still won't work for pixels which are already a floating point value - and therefore already in the correct range

Reply:
Actually, I believe that this code could even replace the two previous cases.

Reply:
https://docs.rs/image/latest/image/enum.DynamicImage.html `ImageRgb32F` and `ImageRgba32F`

Reply:
My mistake, I think I was looking at an outdated documentation. I'll handle these two separately. We can assume the range is already correct, right?

Reply:
Yeah the float ones will already be in the correct range

Reply:
Ok, I realized that there was a call to `grayscale()`, so my added code was useless. So, instead, I added support for `Rgb32f` and `Rgba32f`, and corrected the documentation which was wrong.